test/10_before_test/cibuildwheel_test.py

```diff
@@ -0,0 +1,22 @@
+import os
+import utils
+
+def test():
+    project_dir = os.path.dirname(__file__)
+
+    # build the wheels
+    actual_wheels = utils.cibuildwheel_run(project_dir, add_env={
+        # write python version information to a temporary file, this is
+        # checked in setup.py
+        'CIBW_BEFORE_TEST': '''python -c "import sys; open('/tmp/pythonversion.txt', 'w').write(sys.version)" && python -c "import sys; open('/tmp/pythonexecutable.txt', 'w').write(sys.executable)"''',
+        'CIBW_BEFORE_TEST_WINDOWS': '''python -c "import sys; open('c:\\pythonversion.txt', 'w').write(sys.version)" && python -c "import sys; open('c:\\pythonexecutable.txt', 'w').write(sys.executable)"''',
+        'CIBW_TEST_REQUIRES': 'nose',
```

Comment:
Just curious: why depend on `nose`? Why not just `pytest` (since we're already using that) of Python's own unit tests?

Reply:
@joerick ? I copy it from `CIBW_BEFORE_BUILD`

Reply:
At this point, historical reasons, it could equally be pytest. But there's something quite nice about using something that requires installation, that wouldn't be available in PATH otherwise.

Reply:
Ah, yes, I overlooked that in this case, it's pretty good to have an external dependency, as we can also test `CIBW_TEST_REQUIRES` that way :-)

It's just that apparently `nose` is not getting maintenance anymore.

Reply:
good point! i didn't realise. Might be worth changing to `nose2` at some point!

Reply:
So what you suggest? nose2? https://docs.nose2.io/en/latest/

Reply:
Maybe it's not urgent, though? Maybe we shouldn't bother with it in this PR?

Reply:
ok

Reply:
(Unless you or @joerick think it's a small thing and want to, ofc; I don't know. It just sounded as a future thing. And for our purposes, `nose` is still good enough.)